### PR TITLE
Update go and rust dependencies

### PR DESCRIPTION
**Pull request by jmt-lab**
**Description of changes:**

* Updates go dependencies for host-ctr and ecs-gpu-init
* Updates rust dependencies
* Updates aws-sdk to 1 series in matching pattern to tough
* Updates code to changes inherited from rust dependency updates to minimize duplicate dependency versions
* Changed hash for ring in clarify.toml as they adjusted the copyright years for thirdparty/fiat/LICENSE

**Testing done:**

* Built and launched aws-ecs-2

testsys run will be kicked off in the morning

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by jmt-lab** [comment]
hyper proxy fix, this should hopefully get us going

**Comment by jmt-lab** [comment]
Unit tests passed, good to go

File: sources/host-ctr/go.mod [hunk 1]
```diff
@@ -1,18 +1,20 @@
 module host-ctr
 
-go 1.21
+go 1.21.0
+
+toolchain go1.21.4
```
Comment (yeazelm) [inline sources/host-ctr/go.mod:5]: Was this addition required? Since we take the toolchain from our SDK we might not need it.

File: sources/cfsignal/src/cloudformation.rs [hunk 2]
```diff
@@ -21,7 +22,7 @@ pub async fn signal_resource(
         "Region: {:?} - InstanceID: {:?} - Signal: {:?}",
         region, instance_id, status
     );
-    let config = aws_config::from_env()
+    let config = aws_config::defaults(BehaviorVersion::v2023_11_09())
```
Comment (yeazelm) [inline sources/cfsignal/src/cloudformation.rs:25]: Do we have a reason to lock to this version over latest?

File: sources/clarify.toml [hunk 3]
```diff
@@ -168,7 +168,7 @@ license-files = [
 expression = "MIT AND ISC AND OpenSSL"
 license-files = [
     { path = "LICENSE", hash = 0xbd0eed23 },
-    { path = "third_party/fiat/LICENSE", hash = 0xdad0527d },
+    { path = "third_party/fiat/LICENSE", hash = 0x75829ee2 },
```
Comment (yeazelm) [inline sources/clarify.toml:171]: What was the difference in this license?

File: sources/api/pluto/src/ec2.rs [hunk 4]
```diff
@@ -50,8 +51,9 @@ pub(super) async fn get_private_dns_name(region: &str, instance_id: &str) -> Res
 
     let client = if let Some(https_proxy) = https_proxy {
         let http_client = proxy::setup_http_client(https_proxy, no_proxy)?;
+        let http_client = HyperClientBuilder::new().build(http_client);
```
Comment (yeazelm) [inline sources/api/pluto/src/ec2.rs]: This naming is odd to me, can we shift it to not reuse the name, until I realized you were consuming the first `http_client` I wasn't sure if this was intentional.

File: sources/deny.toml [hunk 5]
```diff
@@ -58,17 +58,15 @@ wildcards = "deny"
...
-    # tungstenite uses a newer version of base64, but many of the
-    # first party crates are still in this version
-    { name = "base64", version = "=0.13" },
-    # AWS SDK for rust uses an older version of fastrand
-    { name = "fastrand", version = "=1.9" },
+    # sdk still uses 0.7.5 but everything in this repo now uses 0.8.2
+    # this can be removed once sdk is updated to 0.8.2
+    { name = "heck", version = "=0.4.1" },
+    { name = "snafu", version = "=0.7.5" },
+    { name = "snafu-derive", version = "=0.7.5" },
```
Comment (yeazelm) [inline sources/deny.toml]: Can we have comments for each one just to be clear which ones go away when? If things get refactored/moved I want them to keep their context going forward.

File: sources/host-ctr/go.mod (hunk 1 above)
Comment (jmt-lab) [reply sources/host-ctr/go.mod:5]: I didn't touch this file manually at all, this is what go did I can fix it though

File: sources/cfsignal/src/cloudformation.rs (hunk 2 above)
Comment (jmt-lab) [reply sources/cfsignal/src/cloudformation.rs:25]: Yes this is to prevent any behavior breakage should the GA sdk change the behaviors there-in on a version tick

File: sources/clarify.toml (hunk 3 above)
Comment (jmt-lab) [reply sources/clarify.toml:171]: The copyright year was bumped to 2020

File: sources/api/pluto/src/ec2.rs (hunk 4 above)
Comment (jmt-lab) [reply sources/api/pluto/src/ec2.rs]: sure

File: sources/deny.toml (hunk 5 above)
Comment (jmt-lab) [reply sources/deny.toml]: They all can go away once the above comment is done (bottlerocket-sdk) they are all from snafu

**Comment by jmt-lab** [comment]
Test Report for migration testing and k8s conformance:

```
aarch64-aws-ecs-2-1-initial                                     Test                             passed                                                       1                              0                               0   f3cba92c                         2024-03-19T23:12:31Z
 aarch64-aws-ecs-2-2-migrate                                     Test                             passed                                                       2                              0                               0   f3cba92c                         2024-03-19T23:13:06Z
 aarch64-aws-ecs-2-3-migrated                                    Test                             passed                                                       1                              0                               0   f3cba92c                         2024-03-19T23:13:14Z
 aarch64-aws-ecs-2-4-migrate                                     Test                             passed                                                       2                              0                               0   f3cba92c                         2024-03-19T23:13:45Z
 aarch64-aws-ecs-2-5-final                                       Test                             passed                                                       1                              0                               0   f3cba92c                         2024-03-19T23:13:58Z
 aarch64-aws-k8s-129-1-initial                                   Test                             passed                                                       5                              0                            7406   f3cba92c                         2024-03-19T23:27:23Z
 aarch64-aws-k8s-129-2-migrate                                   Test                             passed                                                       2                              0                               0   f3cba92c                         2024-03-19T23:27:42Z
 aarch64-aws-k8s-129-3-migrated                                  Test                             passed                                                       5                              0                            7406   f3cba92c                         2024-03-19T23:28:43Z
 aarch64-aws-k8s-129-4-migrate                                   Test                             passed                                                       2                              0                               0   f3cba92c                         2024-03-19T23:29:02Z
 aarch64-aws-k8s-129-5-final                                     Test                             passed                                                       5                              0                            7406   f3cba92c                         2024-03-19T23:29:59Z
 aarch64-aws-k8s-129-ipv6-1-initial                              Test                             passed                                                       5                              0                            7406   f3cba92c                         2024-03-19T23:24:18Z
 aarch64-aws-k8s-129-ipv6-2-migrate                              Test                             passed                                                       2                              0                               0   f3cba92c                         2024-03-19T23:24:36Z
 aarch64-aws-k8s-129-ipv6-3-migrated                             Test                             passed                                                       5                              0                            7406   f3cba92c                         2024-03-19T23:25:38Z
 aarch64-aws-k8s-129-ipv6-4-migrate                              Test                             passed                                                       2                              0                               0   f3cba92c                         2024-03-19T23:26:09Z
 aarch64-aws-k8s-129-ipv6-5-final                                Test                             passed                                                       5                              0                            7406   f3cba92c                         2024-03-19T23:27:09Z
 aarch64-aws-k8s-129-conformance                              Test                             passed                                                     392                              0                            7019   2e5debff                         2024-03-20T05:20:16Z
 aarch64-aws-k8s-129-ipv6-conformance                         Test                             passed                                                     392                              0                            7019   2e5debff                         2024-03-20T05:16:44Z
 ```

**Review by jpculp — APPROVED** [review]

File: sources/api/apiclient/src/exec/terminal.rs [hunk 6]
```diff
@@ -59,7 +60,7 @@ impl Terminal {
 
             // Get the current settings of the user's terminal so we can restore them later.
             let current_termios =
-                tcgetattr(STDOUT_FILENO).context(error::TermAttrSnafu { op: "get" })?;
+                tcgetattr(io::stdout()).context(error::TermAttrSnafu { op: "get" })?;
```
Comment (webern) [inline sources/api/apiclient/src/exec/terminal.rs]: note: test enter-admin-container

File: sources/api/pluto/src/aws.rs [hunk 7]
```diff
@@ -20,13 +20,11 @@ pub(super) enum Error {
 
 type Result<T> = std::result::Result<T, Error>;
 
-async fn sdk_imds_client() -> Result<imds::Client> {
+fn sdk_imds_client() -> imds::Client {
```
Comment (webern) [inline sources/api/pluto/src/aws.rs:23]: Woah, wait  a minute. The IMDS client went from async to blocking?

File: sources/api/pluto/src/proxy.rs [hunk 8]
```diff
@@ -45,7 +46,7 @@ pub(crate) fn fetch_proxy_env() -> (Option<String>, Option<String>) {
 pub(crate) fn setup_http_client(
     https_proxy: String,
     no_proxy: Option<String>,
-) -> Result<impl Into<aws_smithy_client::http_connector::HttpConnector>> {
+) -> Result<ProxyConnector<HttpsConnector<HttpConnector>>> {
```
Comment (webern) [inline sources/api/pluto/src/proxy.rs:49]: Note: test proxy

File: sources/deny.toml [hunk 9]
```diff
@@ -33,42 +37,33 @@ exceptions = [
...
 
 skip = [
-    # several dependencies use an older version of toml
-    { name = "toml", version = "=0.5" },
-    # tungstenite uses a newer version of base64, but many of the
-    # first party crates are still in this version
-    { name = "base64", version = "=0.13" },
-    # AWS SDK for rust uses an older version of fastrand
-    { name = "fastrand", version = "=1.9" },
+    # sdk still uses 0.7.5 but everything in this repo now uses 0.8.2
```
Comment (webern) [inline sources/deny.toml]: "sdk" is ambiguous. Use "settings sdk" everywhere to be more clear.

```suggestion
    # settings sdk still uses 0.7.5 but everything in this repo now uses 0.8.2
```

**Review by webern — APPROVED** [review]

**Comment by jmt-lab** [comment]
Updated per matt's comment.

Proxy tested and passes

* Ran squid proxy on my dev instance
* Ensured squid proxy will forward auth headers and allow passing to amazonaws
* Verified squid logs received and sent requests
* Verified in CloudTrail aws actions seen from k8s instance with source ip of my dev instance (proxy host)

File: sources/api/pluto/src/aws.rs (hunk 7 above)
Comment (jmt-lab) [reply sources/api/pluto/src/aws.rs:23]: yerp

**Review by jpculp — APPROVED** [review]
